WordPress/src/main/java/org/wordpress/android/viewmodel/pages/PageItemUiStateHelper.kt

```diff
@@ -117,6 +132,38 @@ class PageItemUploadProgressHelper @Inject constructor(
         }
     }
 
+    fun setupPageActions(listType: PageListType, pageId: LocalId, site: SiteModel): Set<Action> {
+        return when (listType) {
+            SCHEDULED, PUBLISHED -> mutableSetOf(VIEW_PAGE, SET_PARENT, MOVE_TO_DRAFT, MOVE_TO_TRASH).apply {
+                if (canCancelPendingAutoUpload(pageId, site)) {
+                    add(CANCEL_AUTO_UPLOAD)
+                }
+            }
+            DRAFTS -> mutableSetOf(VIEW_PAGE, SET_PARENT, PUBLISH_NOW, MOVE_TO_TRASH).apply {
+                if (canCancelPendingAutoUpload(pageId, site)) {
+                    add(CANCEL_AUTO_UPLOAD)
+                }
+            }
+            TRASHED -> setOf(MOVE_TO_DRAFT, DELETE_PERMANENTLY)
+        }
+    }
+
+    @VisibleForTesting(otherwise = PRIVATE)
```

Comment:
How do you feel about using this annotation since I am making the method available for testing?

Reply:
I'm not a fan of using these types of annotations. The fact that you need something like this is a code smell - flawed design. 
Moreover, in this case the code smell isn't just a false positive. This class does more than it should. It breaks the SRP. I understand that the main reason is that you were duplicating the design from the PostList. However, I think we might want to consider breaking it down. I actually already broke it down a bit in https://github.com/wordpress-mobile/WordPress-Android/pull/11278. I'd suggest waiting until https://github.com/wordpress-mobile/WordPress-Android/pull/11278 is merged before doing any further changes on this PR. Wdyt?

Reply:
> I'm not a fan of using these types of annotations. The fact that you need something like this is a code smell - flawed design.

Indeed! I am glad I pointed it out because there was something about it that didn't feel right 🙇  

> Moreover, in this case the code smell isn't just a false positive. This class does more than it should. It breaks the SRP. I understand that the main reason is that you were duplicating the design from the PostList. However, I think we might want to consider breaking it down. I actually already broke it down a bit in #11278. I'd suggest waiting until #11278 is merged before doing any further changes on this PR. Wdyt?

Yes. I really love this approach. Looking at the `usecases` you created for the labels and colors has solidified the approach that should be taken for components so they are more maintainable and testable.

Reply:
Removed when I moved this to it's own use case 
aa8231a